pkg/apis/application/v1alpha1/types.go

```diff
@@ -2166,11 +2166,15 @@ type ResourceActionDefinition struct {
 // TODO: describe this type
 // TODO: describe members of this type
 type ResourceAction struct {
-	Name        string                `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
-	Params      []ResourceActionParam `json:"params,omitempty" protobuf:"bytes,2,rep,name=params"`
-	Disabled    bool                  `json:"disabled,omitempty" protobuf:"varint,3,opt,name=disabled"`
-	IconClass   string                `json:"iconClass,omitempty" protobuf:"bytes,4,opt,name=iconClass"`
-	DisplayName string                `json:"displayName,omitempty" protobuf:"bytes,5,opt,name=displayName"`
+	Name          string                `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+	Params        []ResourceActionParam `json:"params,omitempty" protobuf:"bytes,2,rep,name=params"`
```

Comment:
Actually, there is an existing `Params` field, and looking at `ResourceActionParam`, it has most of the needed fields, I checked and it is used in the `util/lua/lua_test.go`, and there is even a test there for an action `scale`. Can you maybe make use of this field and check what this existing `scale` test is used for?